python/cudf/cudf/core/frame.py

```diff
@@ -1821,6 +1829,249 @@ def sample(
 
             return result
 
+    @classmethod
+    @annotate("FROM_ARROW", color="orange", domain="cudf_python")
+    def from_arrow(cls, data):
+        """Convert from PyArrow Table to Frame
+
+        Parameters
+        ----------
+        data : PyArrow Table
+
+        Raises
+        ------
+        TypeError for invalid input type.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> import pyarrow as pa
+        >>> data = pa.table({"a":[1, 2, 3], "b":[4, 5, 6]})
+        >>> cudf.core.frame.Frame.from_arrow(data)
+           a  b
+        0  1  4
+        1  2  5
+        2  3  6
+        """
+
+        if not isinstance(data, (pa.Table)):
+            raise TypeError(
+                "To create a multicolumn cudf data, "
+                "the data should be an arrow Table"
+            )
+
+        column_names = data.column_names
+        pandas_dtypes = None
+        np_dtypes = None
+        if isinstance(data, pa.Table) and isinstance(
+            data.schema.pandas_metadata, dict
+        ):
```

Comment:
Don't need to check if `data` is a `pa.Table` again here since you already confirmed it above.
```suggestion
        if isinstance(data.schema.pandas_metadata, dict):
```